lib/graphql/batch.rb

```diff
@@ -12,20 +12,22 @@ class NestedError < StandardError; end
     def self.batch
       raise NestedError if GraphQL::Batch::Executor.current
       begin
-        GraphQL::Batch::Executor.current = GraphQL::Batch::Executor.new
+        GraphQL::Batch::Executor.current = GraphQL::Batch::Executor.new # TODO
         ::Promise.sync(yield)
       ensure
         GraphQL::Batch::Executor.current = nil
       end
     end
 
-    def self.use(schema_defn)
+    def self.use(schema_defn, executor_class: GraphQL::Batch::Executor)
       schema = schema_defn.target
       if GraphQL::VERSION >= "1.6.0"
+        GraphQL::Batch::SetupMultiplex.executor_class = executor_class
         instrumentation = GraphQL::Batch::SetupMultiplex.new(schema)
         schema_defn.instrument(:multiplex, instrumentation)
         schema_defn.instrument(:field, instrumentation)
       else
+        GraphQL::Batch::Setup.executor_class = executor_class
         instrumentation = GraphQL::Batch::Setup.new(schema)
```

Comment:
Store the state on the instance rather than the class.  It is only supposed to affect the schema it is called on.